ydb/library/yql/dq/opt/bitset.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <stdlib.h>
+
+namespace NYql::NDq {
+
+template <typename TNodeSet>
+inline bool AreOverlaps(const TNodeSet& lhs, const TNodeSet& rhs) {
+    return (lhs & rhs) != 0;
+}
+
+template <typename TNodeSet>
+inline bool IsSubset(const TNodeSet& lhs, const TNodeSet& rhs) {
+    return (lhs & rhs) == lhs;
+}
+
+template <typename TNodeSet>
+inline bool HasSingleBit(TNodeSet nodeSet) {
+    return nodeSet.count() == 1;
+}
+
+template <typename TNodeSet>
+inline size_t GetLowestSetBit(TNodeSet nodeSet) {
+    for (size_t i = 0; i < nodeSet.size(); ++i) {
+        if (nodeSet[i]) {
+            return i;
+        }
+    }
+
+    Y_ASSERT(false);
+    return nodeSet.size();
+}
+
+template <typename TNodeSet>
```

Comment:
Need a comment for this iterator, is it doing anything else than just iterating? And where is it used?